Cargo.toml

```diff
@@ -13,3 +13,6 @@ readme = "README.md"
 
 [package.metadata.docs.rs]
 targets = ["x86_64-unknown-linux-gnu"]
+
+[dependencies]
+retain_mut = "0.1"
```

Comment:
This is a small crate with no dependencies which basically just copy the function in std and change the signature.

If you are unhappy with any dependencies, the code can be rewritten into collecting into a new `Vec`, but that would likely have some performance penalty. Also when [`Vec::drain_filter`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.drain_filter) is stabilized, this wouldn't be needed either.